backend_typescript/models/Game.ts

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    getGameChatManager(): ChatManager {
+        return new ChatManager(this.id);
+    }
+
+    async getGameInfo(reqUserId: number): Promise<Result<pf.Waiting | pf.Winner | pf.PlacingShips | pf.GamePlay>> {
+        let result;
+        try {
+            const gameDetails = await db.getGameById(this.id);
+
+            if (gameDetails.state === 'invited') {
+                return new Failure('Game is in state invited', 400);
+            };
+
+            let gameUser: string;
+            let gamePlayer: number;
+            let gameOpponent: number;
+            if (gameDetails.user1 === reqUserId) {
+                gameUser = 'user1'; //string
+                gamePlayer = gameDetails.user1;
+                gameOpponent = gameDetails.user2; //id
+            } else {
+                gameUser = 'user2'; //string
+                gamePlayer = gameDetails.user2;
+                gameOpponent = gameDetails.user1; //id
+            };
+
+            const opponentsInformation = await db.getUserById(gameOpponent);
+            const playersInformation = await db.getUserById(gamePlayer);
+
+            if (gameDetails.user1 === reqUserId && gameDetails.state === 'user1_ready' || gameDetails.user2 === reqUserId && gameDetails.state === 'user2_ready') {
+                // waiting_for_other_player phase
+                result = pf.waitingForPlayer(gameDetails, opponentsInformation, playersInformation);
+            } else if (gameDetails.state === 'user1_won' || gameDetails.state === 'user2_won') {
+                // winner phase
+                result = pf.winnerPhase(gameDetails, gameUser, opponentsInformation, playersInformation);
+            } else if (gameDetails.state === 'accepted' || (gameDetails.user1 === reqUserId && gameDetails.state === 'user2_ready') || (gameDetails.user2 === reqUserId && gameDetails.state === 'user1_ready')) {
+                // placing_pieces phase
+                result = await pf.placingPieces(gameDetails, this.id, reqUserId, shipsAmount[gameDetails.dimension], opponentsInformation, playersInformation);
+            } else if (gameDetails.state === 'user1_turn' || gameDetails.state === 'user2_turn') {
+                // gamePlay phase
+                result = await pf.gamePlay(gameDetails, this.id, reqUserId, gameUser, gameOpponent, opponentsInformation, playersInformation);
+            };
+
+            return new Success(result);
+        } catch (e) {
+            return new Failure('Server error', 500);
+        }
+    }
+
+    private verifyAcceptDecline(reqUserId: number): string {
+        if (reqUserId !== this.user2) {
+            return 'You are not the correct opponent player';
+        }
+        if (this.state !== 'invited') {
+            return 'Cannot accept or delete an active game';
+        }
+        return 'Action can be made';
+    }
+
+    private async updateGameState(state: string): Promise<Game> {
```

Comment:
please surround content of this function with `try...catch` (db functions can fail) and use return type `Promise<Result<Game>>`